### ISSUE TITLE
problem indexing xml files content

### ISSUE THREAD

**Issue by yermulnik** [dadoonet/fscrawler#163]
I've got xml files with almost the same content like this:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<spocosy version="1.0"><subscription-update subscriptionid="0" requestid="0" last_push="2016-06-03 06:21:34" current_push="2016-06-03 06:21:37" exec="0.002"><lineup id="0" del="no"/></subscription-update></spocosy>
```
When I run `fscrawler` it strips all the tags from content:
```json
10:15:10,969 TRACE [f.p.e.c.f.FsCrawlerImpl] JSon indexed : {
  "content" : "   \n",
  "attachment" : null,
  "meta" : {
    "author" : null,
    "title" : null,
    "date" : null,
    "keywords" : [ ]
  },
[...]
```
If I delete either first or second line of xml, everything goes fine:
```json
10:17:51,928 TRACE [f.p.e.c.f.FsCrawlerImpl] JSon indexed : {
  "content" : "<spocosy version=\"1.0\"><subscription-update subscriptionid=\"0\" requestid=\"0\" last_push=\"2016-06-03 06:21:34\" current_push=\"2016-06-03 06:21:37\" exec=\"0.002\"><lineup id=\"0\" del=\"no\"/></subscription-update></spocosy>\n",
  "attachment" : null,
  "meta" : {
    "author" : null,
    "title" : null,
    "date" : null,
    "keywords" : [ ]
  },
[...]
```
Here's my `fscrawler` config:
```json
{
  "name" : "xml",
  "fs" : {
    "url" : "/tmp/xml",
    "update_rate" : "5s",
    "includes" : [ "*.xml" ],
    "excludes" : null,
    "json_support" : false,
    "filename_as_id" : false,
    "add_filesize" : false,
    "remove_deleted" : false,
    "store_source" : false,
    "indexed_chars" : "100.0%"
  },
  "server" : null,
  "elasticsearch" : {
    "nodes" : [ {
      "host" : "127.0.0.1",
      "port" : 9200
    } ],
    "index" : "xml",
    "type" : "doc",
    "bulk_size" : 0,
    "flush_interval" : null
  }
}
```
Is there anything I've missed or misconfigured?

**Comment by yermulnik** [comment]
Wow! I've turned `json_support` on and everything become fine! Sorry for the noise =)

**Comment by yermulnik** [comment]
Oops, `"json_support" : true` doesn't help.

**Comment by yermulnik** [comment]
Any help would be appreciated. Thanx!

**Comment by dadoonet** [comment]
I saw you issue but don't have time to look at it for now.
Hopefully next week.

**Comment by glaxaco** [comment]
I reproduced your issue, and I think it might be something with Tika. I set a breakpoint immediately after the Tika call, and the parsedContent comes back as "   \n". I don't know that much about Tika but you may want to investigate further there.

**Comment by yermulnik** [comment]
Yup, I've already did that using Tika explicitly and Tika parses and extracts only the data between xml tags. That's why it extracts "empty space". But fscrawler somehow gets the plain data if I remove either string from my xml-files (see my first message in this issue for examples).
The thing is that I need to index all of my xml-files as plain text files - I need fast grep-like search. Is that possible with use of fscrawler and ElasticSearch?

**Comment by dadoonet** [comment]
So I just added a unit test which shows that indeed nothing is extracted by Tika.

This is not something which can be fixed in FS Crawler itself directly unless we add support for a new feature like the one we added to support raw JSON documents.
We could imagine adding a setting `xml_support` like the `json_support` one. If set, we would treat all documents as XML files, will load them using for example Jackson and generate a JSON from that. Basically convert existing XML to JSON on the fly.

I opened #185 for this.
